spec.html

```diff
@@ -32470,23 +32470,36 @@ <h1>Runtime Semantics: RegExpInitialize ( _obj_, _pattern_, _flags_ )</h1>
-            1. If _BMP_ is *true*, then
-              1. Let _pText_ be the sequence of code points resulting from interpreting each of the 16-bit elements of _P_ as a Unicode BMP code point. UTF-16 decoding is not applied to the elements.
-              1. Parse _pText_ using the grammars in <emu-xref href="#sec-patterns"></emu-xref>. The goal symbol for the parse is |Pattern[~U, ~N]|. If the result of parsing contains a |GroupName|, reparse with the goal symbol |Pattern[~U, +N]| and use this result instead. Throw a *SyntaxError* exception if _pText_ did not conform to the grammar, if any elements of _pText_ were not matched by the parse, or if any Early Error conditions exist.
-              1. Let _patternCharacters_ be a List whose elements are the code unit elements of _P_.
+            1. If _F_ contains *"u"*, let _u_ be *true*; else let _u_ be *false*.
+            1. If _u_ is *true*, then
+              1. Let _patternText_ be ! UTF16DecodeString(_P_).
+              1. Let _patternCharacters_ be a List whose elements are the code points of _patternText_.
             1. Else,
-              1. Let _pText_ be ! UTF16DecodeString(_P_).
-              1. Parse _pText_ using the grammars in <emu-xref href="#sec-patterns"></emu-xref>. The goal symbol for the parse is |Pattern[+U, +N]|. Throw a *SyntaxError* exception if _pText_ did not conform to the grammar, if any elements of _pText_ were not matched by the parse, or if any Early Error conditions exist.
-              1. Let _patternCharacters_ be a List whose elements are the code points of _pText_.
+              1. Let _patternText_ be the result of interpreting each of _P_'s 16-bit elements as a Unicode BMP code point. UTF-16 decoding is not applied to the elements.
+              1. Let _patternCharacters_ be a List whose elements are the code unit elements of _P_.
+            1. Let _parseResult_ be ? ParsePattern(_patternText_, _u_).
+            1. Assert: _parseResult_ is a |Pattern| Parse Node.
             1. Set _obj_.[[OriginalSource]] to _P_.
             1. Set _obj_.[[OriginalFlags]] to _F_.
-            1. Set _obj_.[[RegExpMatcher]] to the internal procedure that evaluates the above parse by applying the semantics provided in <emu-xref href="#sec-pattern-semantics"></emu-xref> using _patternCharacters_ as the pattern's List of |SourceCharacter| values and _F_ as the flag parameters.
+            1. Set _obj_.[[RegExpMatcher]] to the internal procedure that evaluates _parseResult_ by applying the semantics provided in <emu-xref href="#sec-pattern-semantics"></emu-xref> using _patternCharacters_ as the pattern's List of |SourceCharacter| values and _F_ as the flag parameters.
             1. Perform ? Set(_obj_, *"lastIndex"*, 0, *true*).
             1. Return _obj_.
           </emu-alg>
         </emu-clause>
 
+        <emu-clause id="sec-parsepattern" aoid="ParsePattern">
+          <h1>Runtime Semantics: ParsePattern ( _patternText_, _u_ )</h1>
```

Comment:
I don't think this can be Runtime Semantics, [which](https://tc39.es/ecma262/#sec-runtime-semantics) "must be called at runtime".

Reply:
I have a dim view of that definition, but I'm fine with changing "Runtime" to "Static" here.

Reply:
Done via force-push.